src/bitcoin-chainstate.cpp

```diff
@@ -0,0 +1,255 @@
+#include <iostream> // for cout and shit
+#include <functional> // for std::function
+#include <filesystem>
+
+#include <node/chainstate.h>          // for LoadChainstate
+#include <init/common.h>              // for SetGlobals, UnsetGlobals
+#include <validation.h>               // for ChainstateManager, InitScriptExecutionCache, StopScriptCheckWorkerThreads, UpdateUncommittedBlockStructures, BlockManager, DEFAULT_CHECKBLOCKS, DEFAULT_CHECKLEVEL
+#include <validationinterface.h>      // for GetMainSignals, cs_main, CMainSignals, RegisterSharedValidationInterface, UnregisterSharedValidationInterface, CValidationInterface
+#include <core_io.h>                  // for DecodeHexBlk
+#include <consensus/validation.h>     // for BlockValidationState
+#include <chainparams.h>              // for Params, SelectParams, CChainParams
+#include <node/blockstorage.h>        // for fReindex
+#include <util/system.h>              // for gArgs, ArgsManager
+#include <util/thread.h>              // for TraceThread
+#include <scheduler.h>                // for CScheduler
+#include <script/sigcache.h>          // for InitSignatureCache
+
+using node::fReindex;
+using node::LoadChainstate;
+using node::VerifyLoadedChainstate;
+
+const extern std::function<std::string(const char*)> G_TRANSLATION_FUN = nullptr;
+
+// Adapted from rpc/mining.cpp
+class submitblock_StateCatcher final : public CValidationInterface
+{
+public:
+    uint256 hash;
+    bool found;
+    BlockValidationState state;
+
+    explicit submitblock_StateCatcher(const uint256 &hashIn) : hash(hashIn), found(false), state() {}
+
+protected:
+    void BlockChecked(const CBlock& block, const BlockValidationState& stateIn) override {
+        if (block.GetHash() != hash)
+            return;
+        found = true;
+        state = stateIn;
+    }
+};
+
+
+int main(int argc, char* argv[]) {
```

Comment:
What would happen if this function was empty? If it builds, then the changes here are minimal and build-system only and wouldn't need to concern itself with (C++) code yet.

Reply:
If it's empty, then we would be able to remove all files in `libbitcoinkernel_la_SOURCES` and it would still compile and link. It being not empty is to enforce that we've linked at least the set of files necessary for this functionality.